src/lightning/pytorch/callbacks/batch_size_finder.py

```diff
@@ -168,31 +168,36 @@ def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: O
                 " If this is not the intended behavior, please remove either one."
             )
 
-    def scale_batch_size(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+    def scale_batch_size(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", is_training: bool) -> None:
         new_size = _scale_batch_size(
             trainer,
             self._mode,
             self._steps_per_trial,
             self._init_val,
             self._max_trials,
             self._batch_arg_name,
+            is_training,
         )
```

Comment:
@tanaymeh On a second thought, I think we can avoid adding a new argument here

`pl_module` passed to `scale_batch_size` anyway, so there are no need to read `pl_module.training` earlier

Example:
```python3
    def scale_batch_size(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        is_training = pl_module.training
        new_size = _scale_batch_size(
            trainer,
            self._mode,
            self._steps_per_trial,
            self._init_val,
            self._max_trials,
            self._batch_arg_name,
            is_training,
        )
...
    def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
        self.scale_batch_size(trainer, pl_module)
...
```

Doing it that way would keep compatibility for people who might have called to `scale_batch_size` in their own code (e.g. classes inherited from `BatchSizeFinder`)